fs/operations/dedupe.go

```diff
@@ -18,16 +18,23 @@ import (
 )
 
 // dedupeRename renames the objs slice to different names
-func dedupeRename(remote string, objs []fs.Object) {
-	f := objs[0].Fs()
-	doMove := f.Features().Move
+func dedupeRename(f fs.Fs, remote string, objs []fs.Object) {
+	info := objs[0].Fs()
+	doMove := info.Features().Move
 	if doMove == nil {
-		log.Fatalf("Fs %v doesn't support Move", f)
+		log.Fatalf("Fs %v doesn't support Move", info)
 	}
 	ext := path.Ext(remote)
 	base := remote[:len(remote)-len(ext)]
 	for i, o := range objs {
-		newName := fmt.Sprintf("%s-%d%s", base, i+1, ext)
+		suffix := 1
+		newName := fmt.Sprintf("%s-%d%s", base, i+suffix, ext)
+		_, err := f.NewObject(newName)
```

Comment:
There are 3 possible returns we are interested in here
- `err == nil` - all good!
- `err == fs.ErrorObjectNotFound` - retry
- `err != nil` - some other error

You are ignoring the `some other error` case.